### PR TITLE
People: Invites: Redirect and display notice on invite dismiss

### LINKED ISSUE #1146: People: Create invitation declined notice
Currently, when a user declines an invitation, we send that user to `?invite_declined`, but we aren't displaying any UI for that or even removing `?invite_declined` from the route.

screen shot 1

### PR DESCRIPTION
**Pull request by lezama**
Fixes #1146

This PR displays a notice when declining and invite and redirects the user to the reader

image

__To test__
- Checkout `update/invites-redirect-and-notice-on-dismiss `

__Decline Invite__
- Create an invite at `$site/wp-admin/users.php?page=wpcom-invite-users`
- In the invite email, make note of the invitation key
- While logged in, go to `/accept-invite/$site/$invite_key`
- Dismiss the notice.
- Do you get redirected to the reader?
- Do you get the dismiss notice?

cc @ebinnion

### CODE REVIEW

File: client/my-sites/invites/invite-accept-logged-in/index.jsx [hunk 1]
```diff
@@ -11,39 +11,36 @@ import Card from 'components/card';
...
 
-const user = userModule();
-
 export default React.createClass( {
 
 	displayName: 'InviteAcceptLoggedIn',
 
 	render() {
-		let userObject = user.get(),
+		const { user } = this.props,
```
Comment (ebinnion) [inline client/my-sites/invites/invite-accept-logged-in/index.jsx]: :+1:

File: client/my-sites/invites/invite-accept/index.jsx [hunk 2]
```diff
@@ -62,17 +63,21 @@ export default React.createClass( {
...
 			default:
-				redirectTo += '/posts/' + invite.blog_id;
+				return '/posts/' + invite.blog_id;
 		}
-		return redirectTo += '?invite_accepted=' + invite.blog_id;
+	},
+
+	decline() {
+		page( '/' );
+		displayInviteDeclined();
```
Comment (ebinnion) [inline client/my-sites/invites/invite-accept/index.jsx]: I noticed a bit of jankiness with this notice. At times the notice will appear on the invite accepted page and then show again on the the reader. I would think that we should only show the notice on the reader. I haven't looked at the implementation behind this, but I recall that we have an argument to show notices on the next page. Perhaps we should use that here?

screen shot 1

**Comment by ebinnion** [comment]
Code-wise, this LGTM. Functionally, I only noticed the one thing about notices.

I wouldn't expect a green notice for declining here. Pinging @rickybanister for a design review and advice.

**Comment by rickybanister** [comment]
I think the blue is-info notice is the correct way to go here.

screen shot 2015-12-02 at 6 31 48 pm

And just to be sure—you'd only be redirected to the Reader if you decline after clicking a link in an email while logged in—that is to say you are redirected to `/`. 

If you click the join or decline button in a notification it would not redirect you anywhere (as in the screenshot above)

**Comment by lezama** [comment]
Thanks! changed it to 'is-info'

> And just to be sure—you'd only be redirected to the Reader if you decline after clicking a link in an email while logged in—that is to say you are redirected to /.

Yes

> If you click the join or decline button in a notification it would not redirect you anywhere (as in the screenshot above)

We are not dealing with notifications right now, but sure.

**Pull request merged**